eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.typed.{ActorRef, Behavior}
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Satoshi}
+import fr.acinq.eclair.channel.{ChannelConfig, LocalParams, SupportedChannelType}
+import fr.acinq.eclair.io.Peer.{ConnectedData, OutgoingMessage, SpawnChannelNonInitiator}
+import fr.acinq.eclair.wire.protocol.{Error, OpenChannel, OpenDualFundedChannel}
+import fr.acinq.eclair.{AcceptOpenChannel, InterceptOpenChannelPlugin, InterceptOpenChannelReceived, InterceptOpenChannelResponse, RejectOpenChannel}
+
+import scala.concurrent.duration.FiniteDuration
+
+// Child actor of a Peer that handles the interactions with an InterceptOpenChannelPlugin:
+//  - sends OpenChannelReceived to the plugin
+//  - waits for its response and translates it to the Peer data format (either SpawnChannelNonInitiator or an outgoing Error)
+//  - reject the channel open after a timeout (configurable in eclair.conf)
```

Comment:
nit: this isn't configurable anymore for simplicity :)

Reply:
Fixed in [7ff2a36](https://github.com/ACINQ/eclair/pull/2552/commits/7ff2a367a33eeddbe9d59b4b07dfc3a7f014b462)